api/collection_test.go

```diff
@@ -623,6 +632,249 @@ func TestPostCollection_invalidRequestBody(t *testing.T) {
+
+			r = mux.SetURLVars(r, map[string]string{
+				"collection_id": invalidCollectionID,
+			})
+
+			api.PutCollectionHandler(w, r)
+
+			Convey("Then the response has the expected status code", func() {
+				So(w.Code, ShouldEqual, http.StatusBadRequest)
+			})
+
+			Convey("Then the response body should contain the expected error response", func() {
+				body, err := ioutil.ReadAll(w.Body)
+				So(err, ShouldBeNil)
+				response := models.ErrorsResponse{}
+				err = json.Unmarshal(body, &response)
+				So(err, ShouldBeNil)
+				So(len(response.Errors), ShouldEqual, 1)
+				So(response.Errors[0].Message, ShouldEqual, collections.ErrInvalidID.Error())
+			})
+		})
+	})
+}
+
+func TestPutCollection_emptyIfMatchHeaderError(t *testing.T) {
+
+	collectionJson := `{
+		"name": "Coronavirus key indicators",
+		"publish_date": "2020-05-05T14:58:29.317Z"
+	}`
+
+	Convey("Given a request to PUT a collection with an empty ", t, func() {
+
+		paginator := mockPaginator()
+		collectionStore := mockCollectionStore()
+
+		r := httptest.NewRequest("PUT", "http://localhost:26000/collections", bytes.NewBufferString(collectionJson))
+		w := httptest.NewRecorder()
+
+		Convey("When the request is sent to the API", func() {
+
+			api := api.Setup(context.Background(), mux.NewRouter(), paginator, collectionStore)
+
+			r = mux.SetURLVars(r, map[string]string{
+				"collection_id": collectionID,
+			})
+
+			api.PutCollectionHandler(w, r)
+
+			Convey("Then the response has the expected status code", func() {
+				So(w.Code, ShouldEqual, http.StatusBadRequest)
+			})
+		})
+	})
+}
+
+func TestPutCollection_noETagInRequest(t *testing.T) {
+
+	collectionJson := `{
+		"name": "Coronavirus key indicators",
+		"publish_date": "2020-05-05T14:58:29.317Z"
+	}`
+
+	Convey("Given a request to PUT a collection with no If-Match header", t, func() {
+
+		paginator := mockPaginator()
+		collectionStore := mockCollectionStore()
+
+		r := httptest.NewRequest("PUT", "http://localhost:26000/collections", bytes.NewBufferString(collectionJson))
+		w := httptest.NewRecorder()
+
+		Convey("When the request is sent to the API", func() {
+
+			api := api.Setup(context.Background(), mux.NewRouter(), paginator, collectionStore)
+
+			r = mux.SetURLVars(r, map[string]string{
+				"collection_id": collectionID,
+			})
+
+			api.PutCollectionHandler(w, r)
+
+			Convey("Then the response has the expected status code", func() {
+				So(w.Code, ShouldEqual, http.StatusBadRequest)
+			})
+
+			Convey("Then the response body should contain the expected error response", func() {
+				body, err := ioutil.ReadAll(w.Body)
+				So(err, ShouldBeNil)
+				response := models.ErrorsResponse{}
+				err = json.Unmarshal(body, &response)
+				So(err, ShouldBeNil)
+				So(len(response.Errors), ShouldEqual, 1)
+				So(response.Errors[0].Message, ShouldEqual, collections.ErrNoIfMatchHeader.Error())
+			})
+		})
+	})
+}
+
+func TestPutCollection_storeError(t *testing.T) {
```

Comment:
Is this test implementation correct for this name? The name suggests an error case but the impl looks like a success test case.